src/mysql_conn.erl

```diff
@@ -169,7 +169,13 @@ connect_socket(#state{tcp_opts = TcpOpts, host = Host, port = Port} = State) ->
     {ok, State#state{socket = Socket}}.
 
 sanitize_tcp_opts(TcpOpts0) ->
-    TcpOpts1 = lists:filter(
+    {FirstOpts, TcpOpts1} = case TcpOpts0 of
```

Comment:
Please add a comment about why this is kept as the first option.

This function is becoming a bit complex. Can it be simplified in some way?

One idea:

```Erlang
sanitize_tcp_opts([{inet_backend, _} = InetBackend | TcpOpts0]) ->
    %% This option must remain the first option in the list
    [InetBackend | sanitize_tcp_opts(TcpOpts0)];
sanitize_tcp_opts(TcpOpts0) ->
   %% ...
```

Reply:
Oh, good idea, I was indeed struggling a bit with that ;) Nice.

As for the explaining comment, I'll see to it.

Reply:
Done :)